src/GitLink/project.json

```diff
@@ -0,0 +1,22 @@
+﻿{
+  "dependencies": {
+    "Catel.Core": "4.3.0",
+    "GitTools.Core": "1.0.0-unstable0021",
+    "ImpromptuInterface": "6.2.2",
+    "LibGit2Sharp": "0.21.0.176",
+    "ReadOnlySourceTree": {
+      "version": "0.1.37-beta",
+      "suppressParent": "none"
+    },
+    "Nerdbank.GitVersioning": {
```

Comment:
So, from a GitVersion perspective...

1. The same
2. The same
3. The same
4 and 5. Depending on how you utilize GitVersion, the same thing can be achieved.  We see GitVersion as a utility for asserting a semantic version number, based on the history of git repository, rather than a tool for updating/generating other files.  I typically use GitVersion in conjunction with something else, i.e. psake, or cake, to orchestrate the build pipeline, and update/generate files as required, based on the asserted version number.